### PR TITLE
chore: add github action and script to automatically sync TLDs and resolver keys

### PR DESCRIPTION
**Pull request by sammyluo**
- add github action and script to automatically sync TLDs and resolver keys

### CODE REVIEW

File: scripts/sync-tld-resolver-keys.mjs [hunk 1]
```diff
@@ -0,0 +1,213 @@
+import fetch from "node-fetch";
```
Comment (sammyluo) [inline scripts/sync-tld-resolver-keys.mjs:1]: This script came from https://github.com/V-Shadbolt/tldsresolverkeys-update-script with slight modifications to be compatible with Github Action

Comment (V-Shadbolt) [reply scripts/sync-tld-resolver-keys.mjs:1]: We know the script works but how are you doing the conversion from what the script produces to what is needed in the NPM package? 

The script produces a JSON file with the necessary information but that information needs to be published as js constants like it currently is in the NPM package.

File: .github/workflows/tld-resolver-keys-sync.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Workflow to sync the latest TLDs and resolver keys
```
Comment (V-Shadbolt) [inline .github/workflows/tld-resolver-keys-sync.yaml:1]: I'm not super familiar with yaml but it looks like this just increments NPM package version numbers and publishes the script output as is?

We'll want to take into account changelog and formatting if we can

File: scripts/sync-tld-resolver-keys.mjs (hunk 1 above)
Comment (sammyluo) [reply scripts/sync-tld-resolver-keys.mjs:1]: I modified your script to write the index.js file directly, and commit the change in the Github Action.

**Review by V-Shadbolt — APPROVED** [review]
Looks good to me! LGTM

File: .github/workflows/tld-resolver-keys-sync.yaml (hunk 2 above)
Comment (sammyluo) [reply .github/workflows/tld-resolver-keys-sync.yaml:1]: Added changelog update

**Pull request merged**